invenio_communities/ext.py

```diff
@@ -65,11 +66,20 @@ def init_app(self, app):
         @app.context_processor
         def record_context_processors():
             """Jinja context processors for communities record integration."""
-            def record_communities(record):
+            def communities_ui_data(record):
                 from invenio_communities.records.api import Record, \
                     RecordCommunitiesCollection
-                return RecordCommunitiesCollection(Record(record, model=record.model))
-            return dict(record_communities=record_communities)
+                record_commmunities = RecordCommunitiesCollection(
+                    Record(record, model=record.model))
+                record_communities_dict = record_commmunities.as_dict()
+                communities_dict = {
+                    record_community.community['id']: record_community.community
```

Comment:
Probably already works since the `Community` is a dict subclass

```suggestion
                    record_community.community['id']: record_community.community.dumps()
```

Also not sure if `community['id']` is the comid or the UUID of the community (and it shouldn't be either actually, but `community.comid.pid_value` should be used probably)